rotkehlchen/chain/manager.py

```diff
@@ -1569,6 +1574,61 @@ def query_ethereum_balances(
         self.query_ethereum_tokens()
         self._add_protocol_balances()
 
+    def query_ethereum_lp_balances(
+            self,
+            balances: Dict[str, Dict[Asset, Balance]],
+    ) -> Optional['AddressToUniswapV3LPBalances']:
+        """Queries all ethereum liquidity pool balances and adds them to the balances.
+        Returns the Uniswap V3 LP balances as they're needed for the NFTs balances.
+        May raise:
+        - RemoteError if any of the calls fail.
+        """
+        module_names: List[Literal['sushiswap', 'balancer']] = ['sushiswap', 'balancer']
+        uniswap = self.get_module('uniswap')
+        uniswap_v3_balances = None
+        if uniswap is not None:
+            uniswap_v3_balances = uniswap.get_v3_balances(
+                addresses=self.queried_addresses_for_module('uniswap'),
+            )
+            uniswap_v2_balances = uniswap.get_balances(
+                addresses=self.queried_addresses_for_module('uniswap'),
+            )
+            self.add_ethereum_lp_to_balance(
+                lp_balance=uniswap_v2_balances,
+                balances=balances,
+            )
+
+        for module_name in module_names:
+            module = self.get_module(module_name)
+            if module is not None:
+                module_lp_balances = module.get_balances(
+                    addresses=self.queried_addresses_for_module(module_name),
+                )
+                self.add_ethereum_lp_to_balance(
+                    lp_balance=module_lp_balances,
+                    balances=balances,
+                )
+
+        return uniswap_v3_balances
+
+    def add_ethereum_lp_to_balance(
+            self,
+            lp_balance: Union['AddressToPoolBalances', 'AddressToLPBalances'],
+            balances: Dict[str, Dict[Asset, Balance]],
+    ) -> None:
+        """Adds the liquidity pool balance(one of Uniswap, Sushiswap or Balancer) to the
+        blockchain balances.
+        """
+        if len(lp_balance) != 0:
```

Comment:
nitpick: don't do that. Return earlier.

This should have been:
```suggestion
        if len(lp_balance) == 0:
            return
```

And below you would get the rest unindented